liqd/adhocracy3.mercator
# refactor error handling rest api

API Changens:

Batch request errors:
   - same error descriptions like without batch (before only 400 BadRequest)

Request errors:
   - all HTTPClientErrors have json body with error description (before only 400, 403, 410 and 500)

Internal Changes:

   - mv request error handling / logging to rest.exceptions (SRP)
   - add json body with description to all HTTPClientErrors Fix #1019
   - make sure all request headers, body, errors entries are logged nicely on error 
   - rm dependency to cornice (reduce complexity)
   - reduce noise in bin/polytester output Fix #1277
   - Fixup: passwords in batch requests may appear in log file  #1407 
   - Fix: arbitrary long request body may appear in log file, max length is now  5000 chars
   - add global test fixture request_ 
   - add global test fixture log to capture log messages
